### ISSUE TITLE
feat(v2): differentiable activated droplet count via adjoint ODE + soft threshold (#45b)

### ISSUE THREAD

**Issue by darothen** [darothen/pyrcel#67]
## Context

This is the differentiable counterpart to #45a (hard-threshold Nd diagnostic in \`_compute_summary\`). That diagnostic is physically accurate but not on the JAX differentiable path. This issue tracks making \`Nd\` gradient-accessible so adjoint sensitivities can be computed.

## Why it matters

The primary scientific value of the differentiable parcel model is computing \`∂output/∂θ\` for inputs like updraft speed, aerosol number, hygroscopicity, etc. \`Nd\` is the most physically relevant output — having its gradient unlocks sensitivity analysis, gradient-based inversion, and optimization of aerosol properties.

## Design

A new integrator-level function \`nd_from_parcel(y0, args, t_end, *, epsilon, ...)\` in \`pyrcel/integrator.py\`, analogous to \`find_smax\`. Key elements:

**1. Sigmoid soft threshold** — replace \`𝟙[r ≥ r_crit]\` with \`σ((r − r_crit) / ε)\`

\`\`\`python
Nd_soft = jnp.sum(Nis * jax.nn.sigmoid((rs_final - r_crits) / epsilon))
\`\`\`

As \`ε → 0\`, \`Nd_soft → Nd\` but gradients vanish for off-threshold particles.
As \`ε → ∞\`, gradients are smooth but \`Nd_soft\` is inaccurate.
A physically motivated default might be \`epsilon = 1e-8\` m (~10 nm, small relative to typical r_crit ~100 nm).

**2. JAX-native Köhler crit** — \`_kohler_crit_approx(T, r_drys, kappas)\` from \`_arg2000.py\` is already vectorizable; use it directly on the final trajectory frame (no \`np.asarray\` crossing).

**3. diffrax adjoint** — use \`RecursiveCheckpointAdjoint()\` for memory-efficient \`O(√N)\` exact gradients through the ODE:

\`\`\`python
sol = dfx.diffeqsolve(
    ...,
    adjoint=dfx.RecursiveCheckpointAdjoint(),
)
\`\`\`

**4. Fixed-horizon integration** (no event) — the event-based termination used by \`find_smax\` complicates adjoints. \`nd_from_parcel\` should run to a fixed \`t_end\` (or \`t_smax + dt_extra\`), making the backward pass clean.

## Intended usage

\`\`\`python
from pyrcel.integrator import nd_from_parcel

# scalar differentiable Nd
Nd = nd_from_parcel(y0, args, t_end, epsilon=1e-8)

# gradient w.r.t. all inputs
dNd_dy0, dNd_dargs = jax.grad(nd_from_parcel, argnums=(0, 1))(y0, args, t_end)
\`\`\`

## Validation

Compare \`nd_from_parcel(..., epsilon → 0)\` against the hard-threshold \`Nd\` from #45a across the same test scenarios. FD gradient check against \`jax.grad\` (similar to MBN2014 gradient tests in \`tests/test_mbn2014.py\`).

## Notes

- The soft Nd and hard Nd are not equal for finite \`epsilon\`; both are physically meaningful but for different purposes (gradient vs. display)
- \`epsilon\` may need to be tunable per problem — a single default may not work across all aerosol size ranges
- Consider a \`jax.lax.cond\`-based epsilon schedule for use in optimization loops
- Depends on #45a for the validation target

## Acceptance criteria

- [ ] \`nd_from_parcel(y0, args, t_end, epsilon)\` in \`pyrcel/integrator.py\`, JIT-compilable
- [ ] \`jax.grad(nd_from_parcel)\` returns finite gradients w.r.t. \`y0\`, \`V\`, \`Ns\`, \`kappas\`
- [ ] FD gradient check passes at \`~1e-2\` relative (limited by ODE tolerance floor, same as MBN2014)
- [ ] Hard-threshold agreement: \`|Nd_soft(ε=1e-10) - Nd_hard| / Nd_hard < 1e-3\` across standard test scenarios
- [ ] JIT + vmap compatible (for ensemble gradient runs)

**Comment by darothen** [comment]
Resolved by 799e083 — differentiable N_d implemented via sigmoid soft-threshold on the activation condition, enabling `jax.grad` through the droplet count. Merged to master via #82.
